tests/jax/systems/mappo_system_test.py

```diff
@@ -1,30 +1,161 @@
 import functools
+from datetime import datetime
+from typing import Any, List
 
-import acme
+import launchpad as lp
 import optax
 import pytest
+from launchpad.launch.test_multi_threading import (
+    address_builder as test_address_builder,
+)
 
+from mava.components.jax.building.distributor import Distributor, DistributorConfig
+from mava.core_jax import SystemBuilder
 from mava.systems.jax import mappo
+from mava.systems.jax.launcher import Launcher, NodeType
 from mava.systems.jax.system import System
+from mava.utils import lp_utils
 from mava.utils.environments import debugging_utils
 from mava.utils.loggers import logger_utils
 
 #########################################################################
 # Full system integration test.
 
 
+class MockLauncher(Launcher):
+    def __init__(
+        self,
+        multi_process: bool,
+        nodes_on_gpu: List = [],
+        sp_trainer_period: int = 10,
+        sp_evaluator_period: int = 10,
+        name: str = "System",
+        terminal: str = "current_terminal",
+    ) -> None:
+        """Create mock launcer component in order to be able to \
+            overwrite the launch method which enables for testing \
+                a full distribured Mava system."""
+        super().__init__(
+            multi_process,
+            nodes_on_gpu,
+            sp_trainer_period,
+            sp_evaluator_period,
+            name,
+            terminal,
+        )
+
+    def launch(self) -> None:
+        """Mock launch method for testing a distributed mava system."""
+        if self._multi_process:
+            local_resources = lp_utils.to_device(
+                program_nodes=self._program.groups.keys(),
+                nodes_on_gpu=[],
+            )
+
+            lp.launch(
+                self._program,
+                launch_type="test_mt",
+                local_resources=local_resources,
+            )
+
+        else:
+            episode = 1
+            executor_steps = 0
+
+            _ = self._node_dict["data_server"]
+            _ = self._node_dict["parameter_server"]
+            executor = self._node_dict["executor"]
+            evaluator = self._node_dict["evaluator"]
+            trainer = self._node_dict["trainer"]
+
+            while True:
+                executor_stats = executor.run_episode_and_log()
+
+                if episode % self._sp_trainer_period == 0:
+                    _ = trainer.step()  # logging done in trainer
+                    print("Performed trainer step.")
+                if episode % self._sp_evaluator_period == 0:
+                    _ = evaluator.run_episode_and_log()
+                    print("Performed evaluator run.")
+
+                print(f"Episode {episode} completed.")
+                episode += 1
+                executor_steps += executor_stats["episode_length"]
+
+
+class MockDistributor(Distributor):
```

Comment:
This is just identical to our standard distributor, barring the call to `MockLauncher` instead of `Launcher`. Perhaps just include a comment referring the reader to where to find the actual code, in case it changes?

Also perhaps add a note in the distributor that people should update this section of the test if they change the distributor code.

Reply:
Please see the discussion above. I have added a comment above `MockDistributor` to explain was is being done, where the original code can be found and that the code should be updated if the `Distributor` code changes.

Reply:
Happy now, thanks!